config/confighttp/clientinfohandler.go

```diff
@@ -53,6 +53,8 @@ func contextWithClient(req *http.Request, includeMetadata bool) context.Context
 		cl.Metadata = client.NewMetadata(req.Header.Clone())
 	}
 
+	cl.Host = req.Host
```

Comment:
Something like:

```go
cl.Metadata["host"] = req.Host
```

Make sure to do something similar for gRPC.

Reply:
I think it's not good idea, because now Metadata equals req.Header
why not to use special property without mutate Metadata(eq headers)?

Reply:
And one more problem
```
// Experimental: *NOTE* this option is subject to change or removal in the future.
    IncludeMetadata bool `mapstructure:"include_metadata,omitempty"`
```

Reply:
>  I think it's not good idea, because now Metadata equals req.Header
> why not to use special property without mutate Metadata(eq headers)?

Because it does not scale and assumes that host is somewhat more relevant than other properties that are part of the HTTP envelope.

Reply:
@jpkrohling I think this shows that we need to define the `Metadata` field semantics better. It is currently not clear whether it is incoming connection transport-specific (which means readers of this field become transport-specific too) or it is transport-agnostic (which requires well-defined, standardized set of keys that all transports should map their metadata too). In the later case "Host" would be one of the standard keys that all transports should use to store the value of the host (whatever the meaning of that host is for that particular transport).

I am not sure what is the right way. Ideally readers of the auth Info should not care how (via what transport) the request was received, so transport-agnosticity is preferable for creating e.g. processors that use auth info. However, I am not sure standardization of keys is really possible across transports.

Reply:
sure, on http2 we already can access to Host header with another key - ":authority"

Reply:
Indeed, we need better documentation for this feature as a whole.